lib/openstudio-standards/weather/stat_file.rb

```diff
@@ -284,13 +284,20 @@ def parse
         degree_day_info.each { |dd_info| parse_dd_info(dd_info) }
 
         # parse design temperatures
+        # TMYx stat file format is different than TMY3, cf #1948
+        is_tmyx = @text.include?("Wind Shelter Factor")
```

Comment:
Detect if TMYx.

Another alternative could be to look for `EnergyPlus Weather Converter Version 2024.06.15` (which appears to be the first lines of the TMYx stat files I looked at)